src/core/mac/mac.cpp

```diff
@@ -1301,6 +1301,7 @@ void Mac::RecordFrameTransmitStatus(const TxFrame &aFrame,
 #if OPENTHREAD_CONFIG_MLE_LINK_METRICS_ENABLE
```

Comment:
Why Subject and Initiator should be differentiated? It should be allowed for one Thread device be both Probing Subject and Probing Initiator, right?

Reply:
@librasungirl, according to the _Table 4-1b. Version value indication of mandatory and optional feature support_:
- Devices that **may** support being Link Metrics Probing **Initiator**: Router, REED, FED, MED, SED, SSED.
- Devices that **must** support being Link Metrics Probing **Subject**: Router, REED.

So I believe it would be beneficial to allow building those features separately for the different device types.

Reply:
Anyway that could be something to address in a different PR.

Reply:
@edmont , thanks for the reminder, and agree with your summary 👍

Reply:
I would address this in another PR.